### DISCUSSION TITLE
unable to use in kaggle

### DISCUSSION THREAD

**Discussion by johnyquest7** [butchland/fastai_xla_extensions#21] [Q&A]
Hi I tried using fastai_xla_extensions, but it errored out. 
Have you used it recently in kaggle?

`!pip install -Uqq fastai --upgrade`
ERROR: kornia 0.3.1 has requirement torch==1.5.0, but you'll have torch 1.7.1 which is incompatible.
ERROR: allennlp 1.0.0 has requirement torch<1.6.0,>=1.5.0, but you'll have torch 1.7.1 which is incompatible.

`!pip install -Uqq cloud-tpu-client==0.10 https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.7-cp36-cp36m-linux_x86_64.whl`
ERROR: torch_xla-1.7-cp36-cp36m-linux_x86_64.whl is not a supported wheel on this platform.

how do I resolve these errors?

**Comment by butchland** [comment]
Please file an issue on the issue tracker. I will check this out later. In the mean time, could you check if you can install the latest version of fastai on kaggle? Also check if you can install PyTorch XLA on kaggle as well? I expect either one or both as the main culprits 😂 for your installation problem (as these are the dependencies needed). If it's either one, I would suggest asking those communities (including kaggle) for possible solutions.

**Comment by tyoc213** [comment]
The way I normally have used it (on colab) is like this

```
!pip install -U fastai --upgrade > /dev/null
VERSION = "20201001"  #"20200515" @param ["1.5" , "20200325", "nightly"]
!curl https://raw.githubusercontent.com/pytorch/xla/master/contrib/scripts/env-setup.py -o pytorch-xla-env-setup.py
!echo "-------------------------------------------------------------------------"
#!TORCH_SHOW_CPP_STACKTRACES=1 python pytorch-xla-env-setup.py --apt-packages libomp5 libopenblas-dev
!python pytorch-xla-env-setup.py  --version $VERSION --apt-packages libomp5 libopenblas-dev > /dev/null
```

Haven't tested it on Kaggle in some time, will test on colab too tomorrow.

**Reply by butchland** [reply]
I think his problem is on kaggle. I haven't tried it on kaggle recently, but I will try it when I can.

**Reply by butchland** [reply]
I opened an issue to track this: https://github.com/butchland/fastai_xla_extensions/issues/25
@johnyquest7 - you can do a watch on this issue if you want to get notified if we find out more (or get it resolved)

**Reply by johnyquest7** [reply]
Thanks. I think it is a kaggle environment issue. Packages available in default environment does not match the requirements. I was not able to change it.

**Reply by tyoc213** [reply]
havent checked, but you can `pip install -U fastai` if that is what you mean with "cant change packages".
